### PR TITLE
Bump versions of mlp/merlin dependency

### PR DESCRIPTION
**Pull request by romanwozniak**
#### Context 
As per title. Both dependencies were set to the initial open-source version of merlin/mlp. Update the repo to use more recent versions of these dependencies.

#### Extra:
Fixed `GET /projects` api endpoint by passing the user-email into authZ enforcer.

### CODE REVIEW

File: api/turing/api/projects_api.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func (c ProjectsController) ListProjects(
 	}
 
 	if c.Authorizer != nil {
-		user, _ := vars.get("user")
+		user := r.Header.Get("User-Email")
```
Comment (romanwozniak) [inline api/turing/api/projects_api.go:25]: User email is populated by the auth-proxy, if it's enabled

File: api/turing/middleware/authorization.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ func (a *Authorizer) FilterAuthorizedProjects(
 		return nil, err
 	}
 
-	var allowedProjects []mlp.Project
+	allowedProjects := make([]mlp.Project, 0)
```
Comment (romanwozniak) [inline api/turing/middleware/authorization.go:88]: Otherwise `allowedProjects` could be `nil`, if  `allowedProjectIds` is empty

Comment (ashwinath) [reply api/turing/middleware/authorization.go:88]: They are actually equavilent. `make([]mlp.Project, 0)` == `var allowedProjects []mlp.Project`. Can try here: https://play.golang.org/p/BZYWfhQnXO4

Comment (romanwozniak) [reply api/turing/middleware/authorization.go:88]: They aren't. Check nil slice vs empty slice in Golang, for example: https://medium.com/@habibridho/golang-nil-vs-empty-slice-87fd51c0a4d

Comment (romanwozniak) [reply api/turing/middleware/authorization.go:88]: What's important, that: 
```go
fmt.Println(nilSlice == nil) // Output: true
fmt.Println(emptySlice == nil) // Output: false
```
So when you try to marshal empty slice into JSON, you'll get `[]`, but marshalling of nil slice will end up with `null`

**Review by ashwinath — APPROVED** [review]

File: api/turing/middleware/authorization.go (hunk 2 above)
Comment (ashwinath) [reply api/turing/middleware/authorization.go:88]: Ah interesting.

**Pull request merged**